diff -up lockdev-1.0.1/Makefile.man8 lockdev-1.0.1/Makefile
--- lockdev-1.0.1/Makefile.man8	2009-09-25 18:36:19.000000000 +0200
+++ lockdev-1.0.1/Makefile	2009-09-25 18:37:00.000000000 +0200
@@ -72,7 +72,9 @@ install_profile:	${static} ${shared}
 
 install_doc:	docs/lockdev.3
 	install -m755 -d	${mandir}/man3
+	install -m755 -d	${mandir}/man8
 	install -m644 docs/lockdev.3	${mandir}/man3
+	install -m644 docs/lockdev.8	${mandir}/man8
 
 install_run:	${shared}
 	install -m755 -d	${libdir}
